### PR TITLE
[#24] modify and delete items

### PR DESCRIPTION
**Pull request by tmachows**

### CODE REVIEW

**Review by pkruczek — APPROVED** [review]

File: src/main/java/com/beerjournal/breweriana/item/ItemService.java [hunk 1]
```diff
@@ -34,4 +34,21 @@ ItemDto addItem(String ownerId, ItemDto itemDto) {
         return itemDto;
     }
 
+    ItemDto deleteItem(String ownerId, String itemId) {
```
Comment (cwikp) [inline src/main/java/com/beerjournal/breweriana/item/ItemService.java:38]: I think you should also check whether user is authorized to delete an item from collection (i.e. user can only delete from his own collection) :P

Same with modify.

Comment (tmachows) [reply src/main/java/com/beerjournal/breweriana/item/ItemService.java:38]: I saw that you implemented it by SecurityUtils, but this approach doesn't look too elegant to me. Don't you know if there isn't any better, more 'declarative' way? E.g. with security configuration?

Comment (cwikp) [reply src/main/java/com/beerjournal/breweriana/item/ItemService.java:38]: Well to honest I only know this approach :P. You could probably implement it a little bit better (e.g. with custom annotation). As for security config I think you can only exclude certain endpoint for groups or authorized users but I don't think there is a possibility to check for example if you own a collection as it requires some business logic :thinking: 

Of course I may be wrong on this, but either way I think we should check with SecurityUtils for now and replace it in the future if we will find a better way :)

Comment (tmachows) [reply src/main/java/com/beerjournal/breweriana/item/ItemService.java:38]: Check added for update and delete. But I suppose that it should also be done for adding items ;)

Comment (cwikp) [reply src/main/java/com/beerjournal/breweriana/item/ItemService.java:38]: Yeah I suppose so :D but I guess it's not really that important for this PR :P

**Review by cwikp — APPROVED** [review]

**Pull request merged**